server/src/main/java/io/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -931,7 +931,8 @@ private void updatePayload(final Handle handle, final DataSegment segment) throw
             return handle
                 .createQuery(
                     StringUtils.format(
-                        "SELECT payload FROM %1$s WHERE dataSource = :dataSource and start >= :start and %2$send%2$s <= :end and used = false",
+                        "SELECT payload FROM %1$s WHERE dataSource = :dataSource and start >= :start "
+                        + "and start <= :end and %2$send%2$s <= :end and used = false",
```

Comment:
"end >= :start" will boost performance when there is only 'end' column index (or in theory if most of segments start at the same time and end always at different time). Otherwise the performance will be the same without "end >= :start" condition. All in all, I don't see a real scenario where 'start' index doesn't exist and 'end' exists, and an additional condition makes a SQL more verbose.